mathics/builtin/stats.py

```diff
@@ -0,0 +1,1743 @@
+            return False
+
+
+class CauchyDistribution(_SympyDistribution):
+    
+    params = (RealParam, PositiveRealParam)
+    sympy_name = 'Cauchy'
+
+
+class ChiDistribution(_SympyDistribution):
+
+    params = (PositiveIntParam)
+    sympy_name = 'Chi'
+
+
+class ChiSquareDistribution(_SympyDistribution):
+    
+    params = (PositiveIntParam)
+    sympy_name = 'ChiSquared'
+
+
+class DiscreteUniformDistribution(_SympyDistribution):
+
+    params = (IntegerParam, IntegerParam)
+    sympy_name = 'DiscreteUniform'
+
+    def check_too(self): #the first param must be smaller than the second
+        par1 = self.params[0].get()
+        par2 = self.params[1].get()
+        if par1 < par2:
+            return True
+        else:
+            return False
+    
+
+class ExponentialDistribution(_SympyDistribution):
+    
+    params = (PositiveRealParam)
+    sympy_name = 'Exponential'
+
+
+class FRatioDistribution(_SympyDistribution):
+
+    params = (PositiveRealParam, PositiveRealParam)
+    sympy_name = 'FDistribution'
+
+
+class GammaDistribution(_SympyDistribution):
+
+    params = (PositiveRealParam, PositiveRealParam)
+    sympy_name = 'Gamma'
+
+
+class LaplaceDistribution(_SympyDistribution):
+
+    params = (RealParam, PositiveRealParam)
+    sympy_name = 'Laplace'
+
+    rules = {
+        'LaplaceDistribution[]': 'LaplaceDistribution[0, 1]'
+    }
+
+
+class LogisticDistribution(_SympyDistribution):
+
+    params = (RealParam, PositiveRealParam)
+    sympy_name = 'Logistic'
+
+    rules = {
+        'LogisticDistribution[]': 'LogisticDistribution[0, 1]'
+    }
+
+
+class LogNormalDistribution(_SympyDistribution):
+
+    params = (RealParam, PositiveRealParam) #for second (sigma) holds sigma^2>0
+    sympy_name = 'LogNormal'
+
+    def check_too(self):
+        sigma = self.params[1].get()
+        if sigma**2 > 0:
+            return True
+        else:
+            return False
+
+
+class MaxwellDistribution(_SympyDistribution):
+
+    params = (PositiveRealParam)
+    sympy_name = 'Maxwell'
+
+
+
+class ParetoDistribution(_SympyDistribution):
+
+    params = (PositiveRealParam, PositiveRealParam)
+    sympy_name = 'Pareto'
+
+
+class PoissonDistribution(_SympyDistribution):
+
+    params = (PositiveRealParam)
+    sympy_name = 'Poisson'
+
+
+class RayleighDistribution(_SympyDistribution):
+
+    params = (PositiveRealParam)
+    sympy_name = 'Rayleigh'
+
+
+class TriangularDistribution(_SympyDistribution):
+
+    params = (RealParam, RealParam, RealParam) #first < second, first<=third<=secnd
+    sympy_name = 'Triangular'
+                                    #HAS LIST AS PARAMETER! 
+    rules = {
+        'TriangularDistribution[]': 'TriangularDistribution[{0, 1}, 1/2]',
+        'TriangularDistribution[{min, max}]': 'TriangularDistribution[{min, max}, (max+min)/2]'
+    }
+
+
+class UniformDistribution(_SympyDistribution):
+
+    params = (RealParam, RealParam) #-infinity<first, first<b<infinity
+    sympy_name = 'Uniform'
+
```

Comment:
How do I get a list of parameters? Using something like 
`params = (ListParam(RealParam, RealParam), RealParam)`? How?